ui/next.config.ts

```diff
@@ -1,7 +1,56 @@
 import type { NextConfig } from "next";
+import path from 'path';
+import fs from 'fs';
+
+// Get the extension name dynamically from the directory structure
+const getExtensionName = () => {
+  try {
+    // Get the current directory
+    const currentDir = __dirname;
+    // Extract the extension name from the path (assuming format: custom_nodes/extension_name/ui)
+    const extensionName = path.basename(path.dirname(currentDir));
+    return extensionName;
+  } catch (error) {
+    console.warn('Failed to get extension name dynamically:', error);
+    // Fallback to the current name
+    return 'ComfyStream';
+  }
+};
+
+const extensionName = getExtensionName();
+
+// Check if we're in development mode
+// This is set by the NEXT_PUBLIC_DEV environment variable in package.json scripts
```

Comment:
@ryanontheinside you use `NEXT_PUBLIC_DEV` instead of the more commonly used `NODE_ENV` due to you developing on Windows?